auth_plugins/duo/plugin.py

```diff
@@ -30,35 +30,46 @@ class DuoPlugin(BotAuthPlugin):
     def __init__(self):
         super().__init__()
 
-        self.client = None
+        self.clients = {}
 
     def setup(self, secrets, **kwargs):
-        if not secrets.get("DUO_IKEY") or not secrets.get("DUO_SKEY") or not secrets.get("DUO_HOST"):
-            raise NoSecretsProvidedError("Must provide secrets to enable authentication.")
+        for variable, secret in secrets.items():
+            if "DUO_" in variable:
+                domain, host, ikey, skey = secret.split(",")
```

Comment:
This seems incorrect. All 3 of the variables must be provided for DUO to work.

I would suggest keeping the old code here.